controllers/articles-controller.js

```diff
@@ -20,11 +21,24 @@ exports.getArticleById = (req, res, next) => {
 };
 
 exports.getArticles = (req, res, next) => {
-  selectArticles()
-    .then((articles) => {
-      return res.status(200).send({ articles });
-    })
-    .catch(next);
+  const { topic } = req.query;
+
+  if (topic) {
+    checkTopicExists(topic)
+      .then(() => {
+        return selectArticles(topic);
+      })
+      .then((articles) => {
+        return res.status(200).send({ articles });
+      })
+      .catch(next);
+  } else {
+    selectArticles()
+      .then((articles) => {
+        return res.status(200).send({ articles });
+      })
+      .catch(next);
+  }
```

Comment:
This conditional logic should be in the model, your controller can send through topic even if it doesn't exist on the query and the model can decide if that is an issue or not. Your model should also handle the check exists function as well. Ideally our controller just asks for the data and sends it back if it exists or goes to catch. All logic should be in the model.